dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -733,4 +721,61 @@ protected TrackedEntityInstance postProcessObject( TrackedEntityInstance tracked
     {
         return (trackedEntityInstance == null || trackedEntityInstance.isDeleted()) ? null : trackedEntityInstance;
     }
+
+    @Override
+    public Optional<TrackedEntityInstance> getTrackedEntityInstanceWithUniqueAttributeValue( TrackedEntityInstanceQueryParams params )

```

Comment:
Any reason this method is in the TEI store and not TEAAttribute store? We dont need to retrieve a TEI, just 1 or 0 rows

Reply:
We need to retrieve a TEI UID. Because the request can be an update and so we need to compare UIDs. If they are the same, that's fine. If they are different, that will cause a duplicate. So we need the UID.

And that is also the reason why it is in the TEI store. Because even though it is a query for attribute uniqueness, at the end we also care about the specific TEI as it is important for us. Can move it to a different store. But then the discussion can be why it is there and not in the TEI store if we fetch TEI UID.

Reply:
Hmm, you can include the TEI uid in the SQL, that way you dont care about returning the uid.

Could you explain why you care about the specific TEI? Why is it important?

Reply:
We are checking uniqueness for an attribute, and we have attribute store, so for me it feels natural to keep in attributestore